### ISSUE TITLE
Consumer selector not working

### ISSUE THREAD

**Issue by Schoksen** [apimeister/tibco-ems-rs#1]
When using a consumer selector only "None" really works, when inserting a String the system takes an unrelated value. 
Could be a C link problem.  

When looking into EMS server with GEMS the selector shows "dest" or "dtype".
When using None its empty, as expected.

inserted into your example queue comsumer and looked into GEMS: 
```rust 

    let consumer = session.queue_consumer(destination,Some(String::from("a=b1"))).unwrap();

    // keep connection open for 20 seconds to see consumer details
    use std::{thread, time};
    thread::sleep(time::Duration::from_millis(20000));

**Comment by JensWalter** [comment]
Hi,
thanks for bringing this issue up.
I think I found the cause, which seems to be related to the lifecycle of the selector string. I updated the library accordingly and published a version 0.1.18 to crates.io. I also added a new sample which I used to verify the selector implementation.
I will do some further testing on this and then close the issue.

**Comment by Schoksen** [comment]
When using a random number generated for the sent message as consumer selector, it now works. 
Thank you!

``` rust
let selector:String = String::from(format!("{}='{}'", header_key, header_value));


[2021-01-08T09:27:46Z INFO  regression] starting engine
[2021-01-08T09:27:46Z DEBUG regression] starting send_msg
[2021-01-08T09:27:46Z DEBUG regression] returning success case
[2021-01-08T09:27:46Z DEBUG regression] Some("a=\'test223\'")
[2021-01-08T09:28:06Z INFO  regression] Original Message Header: {"a": "test223"}
[2021-01-08T09:28:06Z INFO  regression] "HelloWorld"
[2021-01-08T09:28:06Z INFO  regression] New Message Header: {"JMSXDeliveryCount": "1", "MessageID": "ID:EMS-SERVER.4F945FF81D19F:1", "a": "test223"}
[2021-01-08T09:28:06Z INFO  regression] "HelloWorld"`
